pkg/trace/api/api.go

```diff
@@ -709,10 +708,42 @@ func droppedTracesFromHeader(h http.Header, ts *info.TagStats) int64 {
 	return dropped
 }
 
-func getProcessTagsFromHeader(h http.Header) string {
+// todo:raphael cleanup unused methods of extraction once implementation
+// in all tracers is completed
+func getProcessTags(h http.Header, p *pb.TracerPayload) string {
+	// Check if the process tags are not already set in the payload
+	if p.Tags != nil {
+		if ptags, ok := p.Tags[tagProcessTags]; ok {
+			return ptags
+		}
+	}
+	// Check if the process tags are set in the first span
+	span, ok := getFirstSpan(p)
```

Comment:
nit for consistency

```go
if span, ok := getFirstSpan(p); ok {
    // rest...
}
```

Reply:
This does prefers process tags on the tracer payload to the header, would we want that, or to merge them instead?

Is there any scenario where both things might be sent? If there is in the future, we would be disregarding the header and that feels like something that would be confusing.

Reply:
yes that's on purpose, since for a given payload there's only one sender (tracer) it should have only one implementation. I'm pre-shooting multiple of them to avoid losing an agent, one will be picked and I'll cleanup the others after